esmvalcore/preprocessor/_regrid.py

```diff
@@ -369,7 +374,7 @@ def _regional_stock_cube(spec: dict):
         circular=True,
     )
 
-    def add_bounds_from_step(coord, step):
+    def add_bounds_from_step(coord: Coord, step: float) -> np.ndarray:
```

Comment:
Perhaps use `DimCoord | AuxCoord` instead, as `Coord`s are the abstract base class

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2772/commits/6eedca2cef8e2dfb157ffff3e63b51c6251043cb